### DISCUSSION TITLE
Adding tab to model via plugin

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox#15720] [Help Wanted!] [answered]
In my plugin, I have created an view that I would like to attach to the Device model, using the` @register_model_view()` decorator, and  sure enough, I get the tab as expected:
![image](https://github.com/netbox-community/netbox/assets/55629049/f2027cba-ff14-49b8-8eff-e049016f5563)

However, clicking on the tab to get the data, that in this case is from an external source via an API request,  and using the .render() function, presents the data (in my case some markdown) without changing the active tab so to speak. Visually it "overrides" the Device tab.
![image](https://github.com/netbox-community/netbox/assets/55629049/98c1f4f8-0577-4b4d-818a-03530588fd96)

So, everything seems to work as expected, except for this little snag, and I have tried to figure out where the problem is, but my wizardry is yet not at that level

**Comment by candlerb** [comment]
If you can make a small self-contained plugin which demonstrates this behaviour (e.g. display a static page instead of making an API call), then:
1. We can easily reproduce it
2. We can see whether this is a bug in the Netbox plugin framework, or in the way you're using it

**Comment by markkuleinio** [comment, marked as answer]
I will try to make a full public example later, the rough steps for now (this is for Prefix but anyway):

Template:
```django
{% extends 'generic/object.html' %}

{% block content %}
My content
{% endblock content %}
```

View:
```python
@register_model_view(Prefix, name="mytab")
class HelloWorldView(View):
    tab = ViewTab(
        label="My Tab",
        permission="ipam.view_prefix",
    )

    def get(self, request, pk):
        prefix = Prefix.objects.get(pk=pk)
        return render(
            request,
            "my_plugin/tab.html",
            {
                "object": prefix,
                "tab": self.tab,
            },
        )
```

Comments:
- `context["object"]` is needed to be able to populate the contents in the page
- `context["tab"]` makes the tab work

**Reply by ziggekatten** [reply]
There we go!

Passing the `self.tab` to `render() `and it works!

I was sniffing at this somewhat but I never thought it would be that simple

**Reply by candlerb** [reply]
Definitely worth documenting at https://docs.netbox.dev/en/stable/plugins/development/views/#additional-tabs IMO.

**Reply by markkuleinio** [reply]
> I will try to make a full public example later

https://github.com/markkuleinio/netbox-tab-example

**Reply by markkuleinio** [reply]
> Definitely worth documenting at https://docs.netbox.dev/en/stable/plugins/development/views/#additional-tabs IMO.

#15727
